packages/block-library/src/spacer/controls.js

```diff
@@ -41,17 +40,16 @@ function DimensionInput( { label, onChange, isResizing, value = '' } ) {
 	} );
 
 	const handleOnChange = ( unprocessedValue ) => {
-		setTemporaryInput( null );
 		onChange( unprocessedValue );
 	};
 
-	const handleOnBlur = () => {
-		if ( temporaryInput !== null ) {
-			setTemporaryInput( null );
-		}
-	};
-
-	const inputValue = temporaryInput !== null ? temporaryInput : value;
+	const computedValue = useMemo( () => {
```

Comment:
Non-blocking: Just a hunch, but to me `parseQuantityAndUnitFromRawValue()` does not seem computationally taxing enough to offset the overhead of a `useMemo`. Not unless we have to loop through hundreds or maybe thousands of allowed units. (Not saying this warrants a perf benchmark, but just throwing it out there in case you were on the fence 😄)

Reply:
Good point, it does feel like a bit of an earlier optimisation.

I went ahead and remove it in https://github.com/WordPress/gutenberg/pull/39513/commits/1c22f978c04290d09525d5e7cf519ad7bccd17e9